### ISSUE TITLE
Appending text with the SentryFileWriter not possible

### ISSUE THREAD

**Issue by Melle010** [getsentry/sentry-android-gradle-plugin#387]
### Gradle Version

7.3

### AGP Version

7.0.3

### Code Minifier/Optimizer

Proguard

### Version

3.1.5

### Sentry SDK Version

6.4.2

### Steps to Reproduce

Appending a text to an existing text file overwrites the previous line. As a result, our log files always consist of only one (the most recent) log message. While debugging I noticed that `SentryFileWriter` is used instead of Java's `FileWriter`:

<img width="825" alt="Bildschirmfoto 2022-10-13 um 10 08 35" src="https://user-images.githubusercontent.com/25613732/195556438-a1d1419a-49c2-48f6-884c-e5358dae7f4e.png">

You can test it with the following code:

```kotlin
init {
    appendToLogFile("log message 1")
    appendToLogFile("log message 2")
}

private fun appendToLogFile(message: String) {
    runCatching {
        FileWriter(getLogFile(), true).use {
            it.append("$message\n")
            it.flush()
        }
    }.onFailure {
        Timber.e(it, "Could not write to log file")
    }
}
```

### Expected Result

A text file with the following content:

```
log message 1
log message 2
```

### Actual Result

A text file with the following content:

```
log message 2
```

[2022-10-13.txt](https://github.com/getsentry/sentry-android-gradle-plugin/files/9774941/2022-10-13.txt)

**Comment by romtsn** [comment]
Hey thanks for the report and the details, we'll take a look 👍

**Comment by romtsn** [comment]
This is indeed a vary nasty bug 🙈 sorry for the troubles caused. The fix is on its way, and we'll release a new version of the gradle plugin to fix the auto-instrumentation as well. Thanks for the comprehensive report!

**Comment by romtsn** [comment]
hey @Melle010 we've released version 3.2.1 of the gradle plugin, which bumps sentry-android to 6.6.0 (where the actual fix is), please try it out and let us know if you still have issues. Thanks!

**Comment by Melle010** [comment]
Hi @romtsn, thank you for the fix. Yes, with gradle plugin version 3.2.1 and sentry-android version 6.6.0 the file writer works again as expected.
